### ISSUE TITLE
TF1.3 to 1.5 drastic changes (apparently) due to gRPC

### ISSUE THREAD

**Issue by OscarDPan** [tensorflow/tensorflow#17188]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: CentOS Linux release 7.1.1503 (Core)
- **TensorFlow installed from (source or binary)**: pip install
- **TensorFlow version (use command below)**: 1.5
- **Python version**: 2.7.15(with TF 1.5) and 2.7.5(with TF1.3)
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

### Describe the problem
I built a distributed training system with TF1.3 and it works smoothly except that it sporadically freeze. I was informed by @mrry that TF1.5 has an upgrade version of gRPC that fixes a deadlock bug that causes servers to hang.

So I tried to upgrade to 1.5 today but then a lot of unexpected problems came up and not sure if I am missing anything important for 1.5 version. 

TL;DR
Soon after I launch a job, it died right away and all of the time it is the PS showing various errors like "unavailable: OS Error", "Stream removed" and "Transport closed" during sess.run(tf.global_variables_initializer). Also this problem is more likely to happen if I launched more PS.

So, is there any important dependencies I have to check when upgrading from 1.3 to 1.5?

**Comment by angersson** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
